core/src/main/java/com/sequenceiq/cloudbreak/converter/BlueprintRequestToBlueprintConverter.java

```diff
@@ -148,14 +148,18 @@ private void validateHostGroups(JsonNode root) {
             JsonNode hostGroupName = hostGroup.path("name");
             if (isTextPropertyMissing(hostGroupName)) {
                 throw new BadRequestException("Validation error: one of the 'host_groups' has no name.");
-            } else if (!StringUtils.isAlphanumeric(hostGroupName.asText())) {
+            } else if (isInvalidHostgroupName(hostGroupName)) {
                 throw new BadRequestException(String.format("Validation error: '%s' is not a valid host group name. Host group name must be alphanumeric.",
                         hostGroupName.asText()));
             }
             validateComponentsInHostgroup(hostGroup, hostGroupName.asText());
         }
     }
 
+    private boolean isInvalidHostgroupName(JsonNode hostGroupName) {
+        return !StringUtils.isAlphanumeric(hostGroupName.asText().replaceAll("-", ""));
```

Comment:
Not so flexible solution, would be better an extendable solution.

Reply:
I'll move it to blueprint utils